File: Java Programming/Queue.java

//Stack
class Queue
{
    public static void main(String arg[])
    {
        int iRet = 0;

        QueueCode obj = new QueueCode();

        obj.Push(51);
        obj.Push(21);
        obj.Push(11);

        obj.Display();

        iRet = obj.Count();
        System.out.println("Number of elements are : "+iRet);

        obj.Pop();
        obj.Display();

        iRet = obj.Count();
        System.out.println("Number of elements are : "+iRet);

    }
}


class node
{
    public int data;
    public node next;

    public node(int no)
    {
        data = no;
        next = null;
    }
}
class QueueCode
{
    private node first;
    private int Count;

    public QueueCode()
    {
        first = null;
        Count = 0;
    }

    public void Push(int no)
    {
        node newn = new node(no);

        if(first == null)   // LL is empty
        {
            first = newn;
        }
        else
        {
            node temp = first;

            while(temp.next != null)
            {
                temp = temp.next;
            }

            temp.next = newn;
        }
        Count++;
    }
    public void Pop()
    {
        if(first == null)       // LL is empty
        {
            return;
        }
        else if(first.next == null) // LL contains single node
        {
            first = null;
        }
        else        // LL contains more than one node
        {
            first = first.next;
        }
        Count--;
    }
    public void Display()
    {
        node temp = first;

        System.out.println("Elements of the Linked List are : ");

        while(temp != null)
        {
                System.out.print("| "+ temp.data +" |->");
                temp = temp.next;
        }
        System.out.println("null");
    }

    public int Count()
    {
        return Count;
    }
}

File: C++ Programming/Queue.cpp
//Queue
#include<iostream>
using namespace std;

typedef struct node
{
    int data;
    struct node * next;
}NODE ,*PNODE;
/*
typedef struct node NODE
typedef struct node* PNODE
*/
class Queue
{
    private:
        PNODE first;
        
        int Count;
        
    public: 
        Queue();
        void Push(int No);
        void Pop();
        void Display();
        int CountNode();
};
Queue :: Queue()
{
    first = NULL;
    
    Count =0;
}
void Queue :: Push(int no)
{
    PNODE Temp  = first;

    PNODE newn =NULL;
    newn=new(NODE);
    newn->data=no;
    newn->next=NULL;
   

    if(first==NULL)
    {
        first=newn;
    }
    else
    {
        while(Temp->next != NULL)
        {
            Temp = Temp -> next;
        }
        Temp->next = newn;
    }
    Count++;
}

void Queue::Pop()
{
    if(first == NULL)
    {
        return;
    }
    else if(first->next == NULL)
    {
        delete first;
        first = NULL;
    }
    else
    {
        PNODE temp = first;

        first = first->next;
        delete temp;
    }
    Count--;
}
void Queue::Display()
{
    PNODE temp = first;

    cout<<"Elements of the linked list are : \n";
    
    while(temp != NULL) // Type 1
    {
        cout<<"| "<<temp->data<<" | -> ";
        temp = temp ->next;
    }
    cout<<"NULL \n";
}

int Queue::CountNode()
{
    return Count;
}
int main()
{
    Queue obj;
    int iRet = 0;

    obj.Push(111);
    obj.Push(101);
    obj.Push(51);
    obj.Push(21);
    obj.Push(11);

    
    obj.Display();

    iRet = obj.CountNode();
    cout<<"Number of elements in the linked list are : "<<iRet<<"\n";

    obj.Pop();
    obj.Display();

    iRet = obj.CountNode();
    cout<<"Number of elements in the linked list are : "<<iRet<<"\n";

    return 0;
}